csatar-plugins/forms/components/BasicForm.php

```diff
@@ -87,36 +154,70 @@ public function getForms() {
      */
     public function onRun() {
 
-        // Run application
-
-
         // Handle file upload requests
         if ($handler = $this->processUploads()) {
             return $handler;
         }
 
+        $this->getComponentSettings();
+
         // Render frontend
         $this->addCss('/modules/system/assets/ui/storm.css');
         $this->addJs('/modules/system/assets/ui/storm-min.js');
         $this->addJs('/plugins/csatar/forms/assets/vendor/dropzone/dropzone.js');
         $this->addJs('/plugins/csatar/forms/assets/js/uploader.js');
         $this->addJs('/plugins/csatar/forms/assets/js/positionValidationTags.js');
 
-        $model_id = $this->param('model_id', 'new');
-        $this->renderedComponent = $this->createForm($this->getForm(), $model_id);
+        $form = $this->getForm();
+        $this->recordKeyValue = $this->param($this->recordKeyParam);
+
+        if($this->readOnly){
+            $this->renderedComponent = $this->createForm($form, true);
+        }
+
+        if($this->recordKeyValue === $this->createRecordKeyword && !$this->readOnly) {
+            $this->renderedComponent = $this->createForm();
+        }
+
+        if($this->recordKeyValue !== $this->createRecordKeyword && !$this->readOnly && $this->recordActionParam) {
+            $action = $this->param($this->recordActionParam);
+
+            switch ($action) {
+                case $this->actionUpdateKeyword:
+                    $this->renderedComponent = $this->createForm();
+                    break;
+                case $this->actionDeleteKeyword:
+                    $this->renderedComponent = $this->onDelete();
+                    break;
+                default:
+                    $this->renderedComponent = $this->createForm(true);
```

Comment:
Fennebb `$this->createForm($form, true)` van. Biztos nem kell valahol más paramétereket megadni?